addons/editor-theme3/theme3.js

```diff
@@ -393,6 +447,20 @@ export default async function ({ addon, console, msg }) {
   addon.self.addEventListener("disabled", updateColors);
   addon.self.addEventListener("reenabled", updateColors);
 
+  // inject() and overrideColours() are called when changing the Scratch theme, language, or editor mode
+  const oldInject = Blockly.inject;
+  Blockly.inject = function (...args) {
+    const workspace = oldInject.call(this, ...args);
+    updateColors();
```

Comment:
```js
if (Blockly.getMainWorkspace()?.getFlyout?.()) updateColors();
```

Reply:
There's some in-depth discussion about the behavior of `getMainWorkspace()` in the issue: #6212

But for this specific crash, doing nothing if `Blockly.inject()` was called because the "create new custom block" modal has been opened doesn't sound wrong. I don't think this code was actually written to try to cover that case anyways.